src/backend/resourcemanager/communication/rmcomm_RMSEG2RM.c

```diff
@@ -107,8 +107,10 @@ int sendIMAlive(int  *errorcode,
 	if ( res != FUNC_RETURN_OK )
 	{
 		rm_pfree(AsyncCommContext, context);
-		elog(WARNING, "Fail to register asynchronous connection for sending "
-					  "IMAlive message. %d", res);
+		elog(LOG, "failed to register asynchronous connection for sending "
+			      "IMAlive message. %d", res);
+		/* Always switch if fail to register connection here. */
+		switchIMAliveSendingTarget();
```

Comment:
How often is switchIMAliveSendingTarget()? I mean if it is called often, we should have a parameter or bit mask to save the info that standby exists or not to avoid frequent string comparison which is obviously heavier than simple logical operations.